client_code/navigation.py

```diff
@@ -60,16 +60,25 @@ def go_to(target):
     link.raise_event("click")
 
 
+def _reset_links():
+    for link in _links:
+        if isinstance(link.role, str):
+            link.role = [link.role]
+        try:
+            link.role = [r for r in link.role if r != "selected"]
+        except TypeError:
+            link.role = []
```

Comment:
What TypeError are we catching here?

Reply:
Probably OTT but I wanted to handle any possibility that role might be None and therefore not iterable.

Reply:
Happy to get rid of it if you prefer

Reply:
Actually, just testing it and None seems to appear sometimes, so I've put this back in

Reply:
Ha! Yes, because of kwargs, as you say!